test/extended/operators/operators.go

```diff
@@ -74,6 +83,13 @@ var _ = g.Describe("[sig-arch][Early] Managed cluster should", func() {
 			badConditions, missingTypes := surprisingConditions(co)
 			if len(badConditions) > 0 {
 				worstCondition := badConditions[0]
+
+				// kube-apiserver blocks upgrades when feature gates are present.
+				// Allow testing of TechPreviewNoUpgrade clusters by ignoring this condition.
+				if isTechPreview && name == "kube-apiserver" && isKubeAPIUpgradableTechPreviewCondition(worstCondition) {
+					continue
```

Comment:
Does this risk ignoring other failure conditions?

Reply:
This may depend on what `surprisingConditions` returns. I would expect this condition to be relatively low down the list of bad conditions, but it's something I need to check.

If my assumption that this isn't a terrible condition is correct, then another failure should take precedence in assignment of `worstCondition` and we should be safe.

Reply:
Confirmed, Upgradable is the last one it looks for when it's searching the conditions, this should be safe as is, if the available or degraded conditions are not as expected they will take precedence